OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs

```diff
@@ -41,6 +41,24 @@ class NukePowerInfo : SupportPowerInfo, IRulesetLoaded, Requires<BodyOrientation
 		[Desc("Custom palette is a player palette BaseName.")]
 		public readonly bool IsPlayerPalette = false;
 
+		[Desc("Trail animation.")]
```

Comment:
How much work do you think it would be to move all of this into a `NukeLaunchInfo` and then define `Projectile: NukeLaunch` like other normal weapons.

Reply:
After some working on it locally, to put the tl;dr first:

As much as I feared.

While none of the individual changes are particularly hard (except one, more on that later), the diff on `NukePower` would be large, the diff on `NukeLaunch` would be large, the diff on the yamls would be large, and the update rule I'd have to write would be huge, because moving to a proper projectile args means moving half of `NukePower`'s properties.

As icing on the cake, I don't know how I would pass the `NukeLaunch.FractionComplete` back to `NukePower` for the beacon timer.
All in all, the PR would very likely go into mid-3-digit removals and low-4-digit additions. Not saying we shouldn't get this over with at some point, but I don't really feel like investing that much effort right now to get... trails, at least not as long as the aircraft/resupply refactors aren't finished (among other things).

Personally, I'd rather swallow the pill of having to write a slightly larger update rule later on when/if we refactor this, than this PR growing stale/getting closed and not revived until maybe a year later.